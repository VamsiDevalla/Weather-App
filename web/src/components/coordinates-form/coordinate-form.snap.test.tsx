import React from 'react';
import renderer from 'react-test-renderer';
import CustomButton from '../lib/custom-button/custom-button.component';
import CoordinateForm from './coordinate-form.component';

jest.mock('../lib/form-input/form-input.component', () => 'FormInput');
jest.mock('../lib/custom-button/custom-button.component', () => 'CustomButton');

describe('Coordinates Form', () => {
  test('should render properly', () => {
    const coords = { lat: 25, lng: 30 };
    const coordinateForm = renderer.create(
      <CoordinateForm coords={coords} zoom={10} initCoords={coords} setCoordinates={jest.fn()} setZoom={jest.fn()} />,
    );
    expect(coordinateForm.toJSON()).toMatchSnapshot();
  });
  test('should map latitude input to the right handler', () => {
    const coords = { lat: 25, lng: 30 };
    const coordHandler = jest.fn();
    const coordinateForm = renderer.create(
      <CoordinateForm
        coords={coords}
        zoom={10}
        initCoords={coords}
        setCoordinates={coordHandler}
        setZoom={jest.fn()}
      />,
    );
    const latInput = coordinateForm.root.findByProps({ id: 'lat', label: 'Latitude', name: 'latitude' });
    latInput.props.handleChange({ target: { value: 45 } });
    expect(coordHandler).toHaveBeenCalledWith({ lat: 45, lng: 30 });
  });
  test('should map longitude input to the right handler', () => {
    const coords = { lat: 25, lng: 30 };
    const coordHandler = jest.fn();
    const coordinateForm = renderer.create(
      <CoordinateForm
        coords={coords}
        zoom={10}
        initCoords={coords}
        setCoordinates={coordHandler}
        setZoom={jest.fn()}
      />,
    );
    const lngInput = coordinateForm.root.findByProps({ id: 'lng', label: 'Longitude', name: 'longitude' });
    lngInput.props.handleChange({ target: { value: 45 } });
    expect(coordHandler).toHaveBeenCalledWith({ lat: 25, lng: 45 });
  });
  test('should map reset button to the right handler', () => {
    const coords = { lat: 25, lng: 30 };
    const initCoords = { lat: 50, lng: 60 };
    const coordHandler = jest.fn();
    const zoomHandler = jest.fn();
    const coordinateForm = renderer.create(
      <CoordinateForm
        coords={coords}
        zoom={10}
        initCoords={initCoords}
        setCoordinates={coordHandler}
        setZoom={zoomHandler}
      />,
    );
    const resetButton = coordinateForm.root.findByType(CustomButton);
    resetButton.props.onClick();
    expect(coordHandler).toHaveBeenCalledWith({ lat: 50, lng: 60 });
    expect(zoomHandler).toHaveBeenCalledWith(16);
  });
});
