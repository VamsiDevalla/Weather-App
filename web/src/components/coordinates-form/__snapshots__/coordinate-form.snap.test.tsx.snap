// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coordinates Form should render properly 1`] = `
<div
  className="coordinate-form-container"
  data-testid="coordinatesFormContainer"
  role=""
>
  <h2
    aria-label="coordinates form"
    className="title"
    data-testid="coordinatesFormTitle"
    id="coordFormTitle"
    role="heading"
  >
    Coordinates
  </h2>
  <form
    aria-labelledby="coordFormTitle"
    data-testid="coordinatesForm"
  >
    <FormInput
      handleChange={[Function]}
      id="lat"
      label="Latitude"
      name="latitude"
      required={true}
      type="number"
      value="25.00"
    />
    <FormInput
      handleChange={[Function]}
      id="lng"
      label="Longitude"
      name="longitude"
      required={true}
      type="number"
      value="30.00"
    />
    <FormInput
      handleChange={[Function]}
      id="zoom"
      label="Zoom"
      name="zoom"
      required={true}
      type="number"
      value={10}
    />
  </form>
  <CustomButton
    aria-label="reset coordinates"
    onClick={[Function]}
    type="button"
  >
    RESET
  </CustomButton>
</div>
`;
